From 280bbeca4909975e1c4f127c69a1f84e8ec1e437 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Wed, 20 Mar 2013 17:45:51 +0800
Subject: [PATCH 07/34] ENGR00253923 Build: refactor kernelmodules compile
 rule

1. Don't insert kernelmodules compile rule in kernelimage's rule.
move it out.
2. Don't let kernelmodules depend config which is the same with
kernelimage. For it is really depend prebuild kernel. Or it will
corrupt android's multi-job compile env.

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 core/Makefile | 43 +++++++++++++++++++++----------------------
 1 file changed, 21 insertions(+), 22 deletions(-)

diff --git a/core/Makefile b/core/Makefile
index e802e4b..98a4543 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -982,6 +982,26 @@ KERNEL_UIMAGE    := kernel_imx/arch/arm/boot/uImage
 KERNEL_OUT := $(TARGET_OUT_INTERMEDIATES)/KERNEL_OBJ
 KERNEL_HEADERS_INSTALL := $(KERNEL_OUT)/usr
 
+
+.PHONY: $(TARGET_PREBUILT_KERNEL)
+kernelimage: $(TARGET_PREBUILT_KERNEL)
+
+$(KERNEL_CONFIGURE): kernel_imx/arch/arm/configs/$(TARGET_KERNEL_DEFCONF)
+	$(MAKE) -C kernel_imx $(TARGET_KERNEL_DEFCONF) $(KERNEL_ENV)
+
+$(KERNEL_HEADERS_INSTALL): $(KERNEL_CONFIGURE)
+	install -d $(KERNEL_OUT)
+	$(MAKE) -C kernel_imx -j$(HOST_PROCESSOR) O=../$(KERNEL_OUT) $(KERNEL_ENV) headers_install
+
+$(TARGET_PREBUILT_KERNEL): $(KERNEL_CONFIGURE)
+	$(MAKE) -C kernel_imx -j$(HOST_PROCESSOR) uImage $(KERNEL_ENV)
+	install -D $(KERNEL_UIMAGE)  $(PRODUCT_OUT)/uImage
+	install -D $(KERNEL_ZIMAGE)  $(PRODUCT_OUT)/kernel
+
+else
+kernelimage:
+endif
+
 .PHONY: kernelmodules
 
 kernel_modules_src_list :=
@@ -995,8 +1015,7 @@ $(eval kernel_modules_dest_list += $(call append-path,$(PRODUCT_OUT),$(_dest))))
 
 ifneq ($(strip $(kernel_modules_src_list)),)
 
-# kernel modules only depends on kernel config. don't needs kernel have built.
-kernelmodules: $(KERNEL_CONFIGURE) | $(ACP)
+kernelmodules: $(TARGET_PREBUILT_KERNEL) | $(ACP)
 	$(MAKE) -C kernel_imx modules $(KERNEL_ENV)
 	$(hide) $(foreach cf, $(TARGET_KERNEL_MODULES),				\
 	$(eval _src := $(call word-colon,1,$(cf)))				\
@@ -1010,26 +1029,6 @@ kernelmodules: $(KERNEL_CONFIGURE) | $(ACP)
 else
 kernelmodules:
 endif
-
-.PHONY: $(TARGET_PREBUILT_KERNEL)
-kernelimage: $(TARGET_PREBUILT_KERNEL)
-
-$(KERNEL_CONFIGURE): kernel_imx/arch/arm/configs/$(TARGET_KERNEL_DEFCONF)
-	$(MAKE) -C kernel_imx $(TARGET_KERNEL_DEFCONF) $(KERNEL_ENV)
-
-$(KERNEL_HEADERS_INSTALL): $(KERNEL_CONFIGURE)
-	install -d $(KERNEL_OUT)
-	$(MAKE) -C kernel_imx -j$(HOST_PROCESSOR) O=../$(KERNEL_OUT) $(KERNEL_ENV) headers_install
-
-$(TARGET_PREBUILT_KERNEL): $(KERNEL_CONFIGURE)
-	$(MAKE) -C kernel_imx -j$(HOST_PROCESSOR) uImage $(KERNEL_ENV)
-	install -D $(KERNEL_UIMAGE)  $(PRODUCT_OUT)/uImage
-	install -D $(KERNEL_ZIMAGE)  $(PRODUCT_OUT)/kernel
-
-else
-kernelimage:
-endif
-
 # ----------------------------------------------------------------
 # support ubifs rules:
 .PHONY: ubiimagesgenerate
-- 
1.8.0

